managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/MetaStoreImpl.java

```diff
@@ -309,7 +309,11 @@ public void asyncRemoveCursor(String ledgerName, String cursorName, MetaStoreCal
                         callback.operationComplete(null, null);
                         return null;
                     }
-                    SafeRunnable.safeRun(() -> callback.operationFailed(getException(ex)));
+                    try {
+                        callback.operationFailed(getException(ex));
+                    } catch (Exception ex2){
+                        log.error("Unexpected throwable caught when executing callback of async remove cursor", ex2);
+                    }
                     return null;
                 }, executor.chooseThread(ledgerName));
```

Comment:
We should catch the exception thrown by `exceptionallyAsync`

Reply:
As we discuss, we can push another pr to fix that, because all other methods are the same implementation.